<h2>Soal 1: Membuat DataFrame</h2>

- Buatlah dataframe dari objek dictionary!

In [3]:
# Membuat dataframes dari dictionary
import pandas as pd
Data_1 = {'Name': ['Aldiva Wibowo', 'Khairatun Melina', 'Kharka Aldileon', 'Azsyara Aylesia'], 'Age': [20,21,15,14], 'Color': ['Blue', 'Gray', 'Black', 'White']}

df_1 = pd.DataFrame(Data_1)
df_1

Name  Age  Color
0     Aldiva Wibowo   20   Blue
1  Khairatun Melina   21   Gray
2   Kharka Aldileon   15  Black
3   Azsyara Aylesia   14  White

<h2>Soal 2: Membaca data dari file</h2>

Download file bernama [diamonds.csv](https://drive.google.com/uc?export=download&id=1qQiM9utQUThTPq8jk8stPMh_y2zfqYOd), kemudian baca data tersebut dan assign ke dalam suatu variabel kemudian tampilkan DataFrame-nya.

In [4]:
# Membaca data dan memasukannya ke dalam suatu variabel
df_diamond = pd.read_csv('diamonds.csv')
df_diamond

Unnamed: 0  carat        cut color  ... price     x     y     z
0               1   0.23      Ideal     E  ...   326  3.95  3.98  2.43
1               2   0.21    Premium     E  ...   326  3.89  3.84  2.31
2               3   0.23       Good     E  ...   327  4.05  4.07  2.31
3               4   0.29    Premium     I  ...   334  4.20  4.23  2.63
4               5   0.31       Good     J  ...   335  4.34  4.35  2.75
...           ...    ...        ...   ...  ...   ...   ...   ...   ...
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[53940 rows x 11 columns]

Expected Result:

![alt text](https://drive.google.com/uc?id=1BvgM5mJEVfj0fllrjkBfQwn0QIFkry_t)

<h2>Soal 3: Pemeriksaan Data Sederhana</h2>

Di soal no 2 kamu telah membaca data tentang diamonds. pada soal ini cobalah suatu metode dari pandas untuk mengetahui beberapa data pertama dan beberapa data terakhir. kemudian sebutkan nilai dari:
- 10 baris pertama dari column price
- 10 baris terakhir dari column color

tunjukan cara kalian menemukan nilai tersebut di block di bawah ini.

In [7]:
df_diamond.head()

Unnamed: 0  carat      cut color clarity  ...  table  price     x     y     z
0           1   0.23    Ideal     E     SI2  ...   55.0    326  3.95  3.98  2.43
1           2   0.21  Premium     E     SI1  ...   61.0    326  3.89  3.84  2.31
2           3   0.23     Good     E     VS1  ...   65.0    327  4.05  4.07  2.31
3           4   0.29  Premium     I     VS2  ...   58.0    334  4.20  4.23  2.63
4           5   0.31     Good     J     SI2  ...   58.0    335  4.34  4.35  2.75

[5 rows x 11 columns]

In [6]:
df_diamond.tail()

Unnamed: 0  carat        cut color  ... price     x     y     z
53935       53936   0.72      Ideal     D  ...  2757  5.75  5.76  3.50
53936       53937   0.72       Good     D  ...  2757  5.69  5.75  3.61
53937       53938   0.70  Very Good     D  ...  2757  5.66  5.68  3.56
53938       53939   0.86    Premium     H  ...  2757  6.15  6.12  3.74
53939       53940   0.75      Ideal     D  ...  2757  5.83  5.87  3.64

[5 rows x 11 columns]

In [17]:
 df_diamond['price'].head(10)

0    326
1    326
2    327
3    334
4    335
5    336
6    336
7    337
8    337
9    338
Name: price, dtype: int64

In [16]:
df_diamond['color'].tail(10)

53930    E
53931    F
53932    E
53933    E
53934    D
53935    D
53936    D
53937    D
53938    H
53939    D
Name: color, dtype: object

<h2>Soal 4: Practice Deskriptif Statistik</h2>

- Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
- Berapa nilai rata-rata dari column price
- Berapa nilai standar deviasi dari column depth
- Berapa nilai maximum dari column carat

In [21]:
# Gunakan suatu metode dari pandas untuk mengetahui deskriptif statistik dari suatu data
df_diamond[['price','depth','carat']].describe()

price         depth         carat
count  53940.000000  53940.000000  53940.000000
mean    3932.799722     61.749405      0.797940
std     3989.439738      1.432621      0.474011
min      326.000000     43.000000      0.200000
25%      950.000000     61.000000      0.400000
50%     2401.000000     61.800000      0.700000
75%     5324.250000     62.500000      1.040000
max    18823.000000     79.000000      5.010000

Jawab: 

Berapa nilai rata-rata dari column price?
Berapa nilai standar deviasi dari column depth?
Berapa nilai maximum dari column carat?

Nilai rata-rata price adalah 3932.799722	
Nilai standar deviasi depth adalah 1.432621	
Nilai max carat adalah 5.010000
